### ISSUE TITLE
Error if boolean passed as row index or slice?

### ISSUE THREAD

**Issue by benjeffery** [tskit-dev/tskit#2705]
I was recently helping a new user who has R experience who tried `tables.nodes['individual'==0]` which gave the first node as it is equal to `tables.nodes[False]` which gets interpreted as `tables.nodes[0]`. (This is because `isinstance(False, numbers.Integral)` is `True` which was a surprise to me) 

I think it should be a `ValueError` to `__getitem__` with a bool? Or do what numpy does and return `[]`.

**Comment by molpopgen** [comment]
Good idea.

Is there a doc example of what would (should?) work: something like `tables.nodes[np.where('individual' == 0)]`?

**Comment by benjeffery** [comment]
I don't think there is a doc example. The equivalent code would be `tables.nodes[tables.nodes.individual == 0]` but then for most use cases this would be a _very_ inefficient way to access nodes.

**Comment by jeromekelleher** [comment]
Yes, this is a good idea. I can't see any way that (e.g.) raising a warning instead would help (unless I guess people were somehow legitimately doing ``tables.nodes[False]``)?

**Comment by benjeffery** [comment]
Closing for inactivity and labelling "future", please re-open if you plan to work on this.
